src/GitHub.VisualStudio/Services/VSGitExtFactory.cs

```diff
@@ -15,23 +16,24 @@ public class VSGitExtFactory
         static readonly ILogger log = LogManager.ForContext<VSGitExtFactory>();
 
         readonly int vsVersion;
-        readonly IAsyncServiceProvider asyncServiceProvider;
+        readonly IServiceProvider serviceProvider;
 
-        public VSGitExtFactory(int vsVersion, IAsyncServiceProvider asyncServiceProvider)
+        public VSGitExtFactory(int vsVersion, IServiceProvider serviceProvider)
         {
             this.vsVersion = vsVersion;
-            this.asyncServiceProvider = asyncServiceProvider;
+            this.serviceProvider = serviceProvider;
         }
 
         public IVSGitExt Create()
         {
             switch (vsVersion)
             {
                 case 14:
-                    return Create(() => new VSGitExt14(asyncServiceProvider));
+                    return Create(() => new VSGitExt14(serviceProvider));
                 case 15:
+                    return Create(() => new VSGitExt15(serviceProvider));
                 case 16:
-                    return Create(() => new VSGitExt15(asyncServiceProvider));
+                    return Create(() => new VSGitExt16(serviceProvider));
                 default:
                     log.Error("There is no IVSGitExt implementation for DTE version {Version}", vsVersion);
                     return null;
```

Comment:
Returning `null` for a MEF service can cause unexpected behavior and isn't a good idea. We should throw here instead.